backend/cmd/headlamp.go

```diff
@@ -1384,3 +1388,122 @@ func absPath(path string) (string, error) {
 
 	return filepath.Join(currentUser.HomeDir, path[2:]), nil
 }
+
+//nolint:funlen
+func (c *HeadlampConfig) handleNodeDrain(w http.ResponseWriter, r *http.Request) {
+	var drainPayload struct {
+		Cluster  string `json:"cluster"`
+		NodeName string `json:"nodeName"`
+	}
+	// get nodeName and cluster from request body
+	err := json.NewDecoder(r.Body).Decode(&drainPayload)
+	if err != nil {
+		http.Error(w, "Error decoding payload", http.StatusBadRequest)
+		return
+	}
```

Comment:
Can you please remove `error` keyword from the error message. For context please refer https://github.com/headlamp-k8s/headlamp/pull/1129#discussion_r1193068710

```suggestion
        if err := json.NewDecoder(r.Body).Decode(&drainPayload); err != nil {
		http.Error(w, "Decoding payload", http.StatusBadRequest)
		return
	}
```